WMF Framework/ArticleCacheController.swift

```diff
@@ -56,9 +64,51 @@ public final class ArticleCacheController: CacheController {
                 completionHandler(error)
                 //tonitodo: broadcast migration error
             }
-        }) { (error) in
-            completionHandler(error)
-            //tonitodo: broadcast migration error
+        }
+    }
+    
+    private func bundledResourcesAreCached(desktopArticleURL: URL) -> Bool {
+        guard let articleDBWriter = dbWriter as? ArticleCacheDBWriter else {
+            return false
+        }
+        
+        return articleDBWriter.bundledResourcesAreCached(desktopArticleURL: desktopArticleURL)
+    }
+    
+    private func cacheBundledResourcesIfNeeded(desktopArticleURL: URL, completionHandler: @escaping ((Error?) -> Void)) { //articleURL should be desktopURL
+        
+        guard let articleDBWriter = dbWriter as? ArticleCacheDBWriter else {
+            completionHandler(CacheFromMigrationError.invalidDBWriterType)
+            return
+        }
+        
+        //tonitodo: this will bundle those shared across siteURLs over and over again. need to bundle shared resources first as a separate thing, then bundle site-specific resource.
```

Comment:
Might be able to skip the site specific CSS depending on how bad migrated articles look without it

Reply:
@joewalsh this looked the same to me without it so I removed the site-specific CSS.